qiita_ware/ebi.py

```diff
@@ -616,57 +608,35 @@ def generate_curl_command(
 
         return curl_command
 
-    def send_sequences(self):
-        # Send the sequence files by directory
-        unique_dirs = set()
-        for f in self.sequence_files:
-            basedir, filename = split(f)
-            unique_dirs.add(basedir)
-
-        # Set the ASCP password to the one in the Qiita config, but remember
-        # the old pass so that we can politely reset it
-        old_ascp_pass = environ.get('ASPERA_SCP_PASS', '')
-        environ['ASPERA_SCP_PASS'] = qiita_config.ebi_seq_xfer_pass
-
-        for unique_dir in unique_dirs:
-            # Get the list of FASTQ files to submit
-            fastqs = glob(join(unique_dir, '*.fastq.gz'))
-
-            ascp_command = 'ascp -d -QT -k2 -L- {0} {1}@{2}:./{3}/'.format(
-                ' '.join(fastqs), qiita_config.ebi_seq_xfer_user,
-                qiita_config.ebi_seq_xfer_url, self.ebi_dir)
-
-            # Generate the command using shlex.split so that we don't have to
-            # pass shell=True to subprocess.call
-            ascp_command_parts = shsplit(ascp_command)
-
-            # Don't leave the password lingering in the environment if there
-            # is any error
-            try:
-                call(ascp_command_parts)
-            finally:
-                environ['ASPERA_SCP_PASS'] = old_ascp_pass
-
-    def send_xml(self):
-        # Send the XML files
-        curl_command = self.generate_curl_command()
-        curl_command_parts = shsplit(curl_command)
-        temp_fd, temp_fp = mkstemp()
-        call(curl_command_parts, stdout=temp_fd)
-        close(temp_fd)
-
-        with open(temp_fp, 'U') as curl_output_f:
-            curl_result = curl_output_f.read()
+    def generate_send_sequences_cmd(self):
+        """Generate the sequences to EBI via ascp command
+
+        Notes
+        -----
+        - All 5 XML files (study, sample, experiment, run, and submission) must
+          be generated before executing this function
+        """
+        fastqs = [sfp for _, sfp in viewitems(self.sample_demux_fps)]
+
+        ascp_command = ('ascp --ignore-host-key -L- -d -QT -k2 '
+                        '{0} {1}@{2}:./{3}/'.format(
+                            ' '.join(fastqs), qiita_config.ebi_seq_xfer_user,
```

Comment:
This will cause issues if submitting a very large number of fastqs due to the command length, but I don't know how best to work around it unless EBI would accept multiple `ascp` calls.